pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -43,6 +44,8 @@ import (
 
 const ResourceResyncTime = 60 * time.Second
 
+const OrphanFinalizerID = "orphanFinalizer"
```

Comment:
No, we can't hard-code them in the api because that would prevent people
from creating their own finalizers, which is a use case we want to support.
In the future we could make a registration step but that deserves its own
design...

On Wed, May 18, 2016 at 4:24 PM, Derek Carr <notifications@github.com>
wrote:

> In pkg/controller/garbagecollector/garbagecollector.go
> <https://github.com/kubernetes/kubernetes/pull/25599#discussion_r63800472>
> :
>
> > @@ -43,6 +44,8 @@ import (
> >
> >  const ResourceResyncTime = 60 * time.Second
> >
> > +const OrphanFinalizerID = "orphanFinalizer"
>
> Can we keep core finalizers like this in the API similar to how we handle
> resource names? That way we can ensure custom finalizers are qualified
> names in validation?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25599/files/7b7917ddcb2f2ab0d8c22748df3668346ea69505#r63800472>
>